File: Web-SPA/src/app/entities/special-offer.ts
import { Flight } from './flight';
import { Trip } from './trip';
import { Seat } from './seat';

export class SpecialOffer {
    flights: Array<Flight>;
    seats: Array<Seat>;
    newPrice: number;

    constructor(flights?: Array<Flight>, seats?: Array<Seat>, newPrice?: number) {
        this.flights = flights || new Array<Flight>();
        this.seats = seats || null;
        this.newPrice = newPrice || null;
    }
}

File: Web2-Microservices/AirlineMicroservice/Models/SpecialOffer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace AirlineMicroservice.Models
{
    public class SpecialOffer
    {
        public int SpecialOfferId { get; set; }
        public float OldPrice { get; set; }
        public float NewPrice { get; set; }
        public bool IsReserved { get; set; }
        public Airline Airline { get; set; }
        public ICollection<Seat> Seats { get; set; }
        public SpecialOffer()
        {
            Seats = new HashSet<Seat>();
            IsReserved = false;
        }
    }
}
